src/test/cs/Infinispan/HotRod/RemoteQueryTest.cs

```diff
@@ -25,42 +34,495 @@ public void BeforeClass()
             if (metadataCache.ContainsKey(ERRORS_KEY_SUFFIX))
             {
                 Console.WriteLine("fail: error in registering .proto model");
                 Environment.Exit(-1);
             }
 
-            User user1 = new User();
-            user1.Id = 4;
-            user1.Name = "Jerry";
-            user1.Surname = "Mouse";
+            IRemoteCache<int, User> userCache = remoteManager.GetCache<int, User>(NAMED_CACHE);
+            PutUsers(userCache);
+            IRemoteCache<int, Account> accountCache = remoteManager.GetCache<int, Account>(NAMED_CACHE);
+            PutAccounts(accountCache);
+            IRemoteCache<int, Transaction> transactionCache = remoteManager.GetCache<int, Transaction>(NAMED_CACHE);
+            PutTransactions(transactionCache);
+        }
 
-            User ret = testCache.Put(4, user1);
+        [Test]
+        public void GetAllTest()
+        {
+            IRemoteCache<int, User> userCache = remoteManager.GetCache<int, User>(NAMED_CACHE);
 
             QueryRequest qr = new QueryRequest();
             qr.JpqlString = "from sample_bank_account.User";
 
-            QueryResponse result = testCache.Query(qr);
-            List<User> listOfUsers = new List<User>();
-            unwrapResults(result, listOfUsers);
+            QueryResponse result = userCache.Query(qr);
+            List<User> listOfUsers = unwrapResults<User>(result);
+
+            Assert.AreEqual(3, listOfUsers.Count);
+        }
+
+        [Test]
+        public void Eq1Test()
+        {
+            IRemoteCache<int, User> userCache = remoteManager.GetCache<int, User>(NAMED_CACHE);
+
+            QueryRequest qr = new QueryRequest();
+            qr.JpqlString = "from sample_bank_account.User u where u.name = \"John\"";
+
+            QueryResponse result = userCache.Query(qr);
+            List<User> listOfUsers = unwrapResults<User>(result);
+
+            Assert.AreEqual(1, listOfUsers.Count);
+            Assert.AreEqual("John", listOfUsers.ElementAt(0).Name);
+            Assert.AreEqual("Doe", listOfUsers.ElementAt(0).Surname);
+        }
+
+        [Test]
+        public void EqEmptyStringTest()
+        {
+            IRemoteCache<int, User> userCache = remoteManager.GetCache<int, User>(NAMED_CACHE);
+
+            QueryRequest qr = new QueryRequest();
+            qr.JpqlString = "from sample_bank_account.User u where u.name = \"\"";
+
+            QueryResponse result = userCache.Query(qr);
+            List<User> listOfUsers = unwrapResults<User>(result);
+
+            Assert.AreEqual(0, listOfUsers.Count);
+        }
+
+        [Test]
+        public void EqSentenceTest()
+        {
+            IRemoteCache<int, Account> accountCache = remoteManager.GetCache<int, Account>(NAMED_CACHE);
+
+            QueryRequest qr = new QueryRequest();
+            qr.JpqlString = "from sample_bank_account.Account a where a.description = \"John Doe's first bank account\"";

```

Comment:
You could use single quote instead of double quote, so it will not have to be escaped.